src/app/reporting/Engine.cpp

```diff
@@ -326,7 +369,11 @@ CHIP_ERROR Engine::BuildSingleReportDataEventReports(ReportDataMessage::Builder
         SuccessOrExit(err = aReportDataBuilder.GetError());
         VerifyOrExit(eventReportIBs.GetWriter() != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
         SuccessOrExit(err = eventReportIBs.GetWriter()->ReserveBuffer(kReservedSizeEndOfReportIBs));
-        err = eventManager.FetchEventsSince(*(eventReportIBs.GetWriter()), eventList, eventMin, eventCount,
+
+        err = CheckAccessDeniedEventPaths(*(eventReportIBs.GetWriter()), hasEncodedData, apReadHandler);
```

Comment:
Isn't this going to keep emitting these StatusIBs in every report, priming or not priming in a subscription? That doesn't seem right.

Reply:
Isn't that the behavior we want?  Spec is not very clear about the non-priming bits....